app/Services/MarketDataProviders/CoinGecko.php

```diff
@@ -104,6 +106,45 @@ public function priceAndPriceChange(string $baseCurrency, Collection $targetCurr
             ->mapWithKeys(fn (string $currency) => [strtoupper($currency) => MarketData::fromCoinGeckoApiResponse($currency, $data)]);
     }
 
+    /**
+     * @return array{
+     *   price: float,
+     *   volume: float,
+     * }|null
+     */
+    public function exchangeDetails(Exchange $exchange): array | null
+    {
+        $data = null;
+
+        try {
+            $data = Http::get('https://api.coingecko.com/api/v3/exchanges/'.$exchange->coingecko_id.'/tickers', [
+                'coin_ids' => 'ark',
+            ])->json('tickers');
+        } catch (\Throwable) {
+            //
+        }
+
+        if ($this->isEmptyResponse($data) || $this->isThrottledResponse($data)) {
+            throw new ApiNotAvailableException();
```

Comment:
You can remove this exception and return null. the `isEmptyResponse` and `isThrottledResponse` methods handle throwing exceptions to reduce spamming exceptions to sentry. On the back of this, you could just not update the values if the response is null, effectively acting as a "forever" cache

Reply:
@alexbarnsley I intended to force the job to run again after one minute, didnt consider the sentry spam so I considered this alternative:

https://github.com/ArdentHQ/arkscan/pull/237/commits/40070b761d8badfcc1b7a0235c9da448a5f45858


Basically throw exception for case the job was throttled and release the job again in 1 min in that case